tests/std/tests/P0896R4_ranges_alg_find/test.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+//
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::find, ranges::iterator_t, std::same_as;
+    using P = std::pair<int, int>;
+
+    // Validate dangling story
+    STATIC_ASSERT(same_as<decltype(find(borrowed<false>{}, 42)), ranges::dangling>);
+    STATIC_ASSERT(same_as<decltype(find(borrowed<true>{}, 42)), int*>);
+
+    std::array<P, 3> const data = {{{0, 42}, {2, 42}, {4, 42}}};
+
+    // Validate range overload
+    for (auto [value, _] : data) {

```

Comment:
If Stephan were here he would say "I want auto&&"

Reply:
The copy here is, uh, intentional to avoid the algorithm taking advantage of the fact that the target element and value have the same address. Yeah, something like that.

Reply:
I believe you, but I am gullible